### PR TITLE
Add nameable and decoration files

### PR DESCRIPTION
**Pull request by Divinity-dev**
In this phase of the project, I have implemented the following;
1. Created a class called nameable and implemented a correct_name method to raise a NotImplementedError
2. Make the person class inherit from the nameable class
3. Created a class called Decorator to inherit the nameable class
4. Created a  CapitalizeDecorator to inherit the decorator class
5. Created a TrimmerDecorator to inherit from the decorator class 
6.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
```
Comment (brytebee) [inline trimmer_decorator.rb:4]: - [ ] **Required:** This method should return only the first 9 characters if a name is longer than that. It does work as required. Consider the code snippet below.

```
def correct_name
   @nameable.correct_name[0..9] 
end
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Divinity-dev 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Divinity-dev** [comment]
Thanks for the correction. I'm implementing right away

Sent from Yahoo Mail on Android 
 
  On Tue, Apr 18, 2023 at 10:03 PM, Bright ***@***.***> wrote:   


@brytebee requested changes on this pull request.

Hi, @Divinity-dev 👋 ,



Highlights 💡
    
   - Good commit message. ✔️
   - Good PR Summary. ✔️
   - Great UI design. ✔️
   - App functions as it should. ✔️



Good job so far coming so far with the progress you've made! 💪 💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



Required Changes ♻️
 
Check the comments under the review.

Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding! 👏 👏 👏
 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.

Use @brytebee on slack for faster responses.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.

In trimmer_decorator.rb:
> @@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
   
   -  Required: This method should return only the first 9 characters if a name is longer than that. It does work as required. Consider the code snippet below.
def correct_name
   @nameable.correct_name[0..9] 
end

**Review by brytebee — APPROVED** [review]
Hi, @Divinity-dev 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Divinity-dev** [comment]
Thank you 

Sent from Yahoo Mail on Android 
 
  On Tue, Apr 18, 2023 at 10:14 PM, Bright ***@***.***> wrote:   


@brytebee approved this pull request.

Hi, @Divinity-dev 👋 ,



Highlights 💡
    
   - Good commit message. ✔️
   - Good PR Summary. ✔️
   - Great UI design. ✔️
   - App functions as it should. ✔️



STATUS: APPROVED 🕺 🥳
 
Congratulations 🍾 🥳 !! Your project is ready for presentation! There is nothing else to say other than... it's time to merge it 



Good luck! 🎉
 


Optional Suggestions:
   
   - N/A

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.

Use @brytebee on slack for faster responses.

As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.

**Pull request merged**